### PR TITLE
Expose FileOutputByteStream's close API into WritableByteStream protocol

### PR DESCRIPTION
**Pull request by sergiocampama**

### CODE REVIEW

**Comment by sergiocampama** [comment]
@swift-ci please test

**Comment by sergiocampama** [comment]
@swift-ci please test

**Comment by sergiocampama** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test

**Review by neonichu — APPROVED** [review]

File: Sources/TSCBasic/WritableByteStream.swift [hunk 1]
```diff
@@ -625,19 +648,24 @@ public final class BufferedOutputByteStream: _WritableByteStreamBase {
...
     }
+
+    override final func closeImpl() throws {
+        // Do nothing. The protocol does not require to stop receiving writes, close only signals that resources could
+        // be released at this point should we need to.
+    }
 }
 
 /// Represents a stream which is backed to a file. Not for instantiating.
 public class FileOutputByteStream: _WritableByteStreamBase {
```
Comment (abertelrud) [inline Sources/TSCBasic/WritableByteStream.swift:659]: Would this at some point be renamed `FileWritableByteStream` or something similar?

**Review by abertelrud — APPROVED** [review]

File: Sources/TSCBasic/WritableByteStream.swift (hunk 1 above)
Comment (sergiocampama) [reply Sources/TSCBasic/WritableByteStream.swift:659]: potentially? not super concerned. The protocol is more important IMO

**Pull request merged**